File: content/posts/Post021 - Partial Regression Plots/partial_regression.R
library(faraway)
library(ggplot2)

data(sat)

# partial regression plot
ggplot(data=sat, aes(x=expend, y=total)) + geom_point()

cor(sat$expend, sat$salary)

expend_resid = resid(lm(data=sat, expend ~ ratio + salary + takers))
total_resid = resid(lm(data=sat, total ~ ratio + salary + takers))

cor(expend_resid, total_resid)

ggplot(data=NULL) + geom_point(aes(x=expend_resid, y=total_resid))


# Partial residual plot
model = lm(data=sat, total ~ expend + ratio + salary + takers)

File: content/posts/Post021 - Partial Regression Plots/partial_regression.jl
using Cairo
using CSV
using DataFrames
using Fontconfig
using Gadfly
using GLM

sat = CSV.File("sat.csv") |> DataFrame
expend_regression = lm(@formula(expend ~ ratio + salary + takers), sat)
total_regression = lm(@formula(total ~ ratio + salary + takers), sat)

expend_residuals = sat["expend"] - predict(expend_regression, sat)
total_residuals = sat["total"] - predict(total_regression, sat)

residual_df = DataFrame(ExpendResiduals=expend_residuals, TotalResiduals=total_residuals)

p = plot(residual_df, x="ExpendResiduals", y="TotalResiduals", Geom.point, Theme(background_color="white"));
img = PNG("julia.png", 8inch, 5inch)
draw(img, p)